### PR TITLE
Some cleanup

### PR DESCRIPTION
**Pull request by davidfowl**
- Use a cached Task to represent true and false
- Remove nested classes around session affinity
- Use nameof for SessionAffinityConstants
- Updated calling code and tests

PS: this PR is mostly cosmetic but the nested classes have been bugging me for a long time. There's more but I was looking at the session affinity logic.

### CODE REVIEW

File: src/ReverseProxy/Utilities/TaskUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.ReverseProxy.Utilities
+{
+    internal static class TaskUtilities
+    {
+        internal static readonly Task<bool> TrueTask = Task.FromResult(true);
```
Comment (davidfowl) [inline src/ReverseProxy/Utilities/TaskUtilities.cs:14]: @stephentoub I wish we had this in the box 😄

Comment (stephentoub) [reply src/ReverseProxy/Utilities/TaskUtilities.cs:14]: https://github.com/dotnet/runtime/issues/22627

Comment (davidfowl) [reply src/ReverseProxy/Utilities/TaskUtilities.cs:14]: FromResult returning a cached task would fix lots of code automagically.

Comment (stephentoub) [reply src/ReverseProxy/Utilities/TaskUtilities.cs:14]: Right, that's my preference as well, at least for the bools.

**Review by ManickaP — APPROVED** [review]

**Review by Tratcher — APPROVED** [review]

**Pull request merged**